collector/exec_bsd.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build freebsd dragonfly
+// +build !noexec
+
+package collector
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type execCollector struct {
+	sysctls []bsdSysctl
+}
+
+func init() {
+	Factories["exec"] = NewExecCollector
+}
+
+// Returns a new Collector exposing system execution statistics
+func NewExecCollector() (Collector, error) {
+	// from sys/vm/vm_meter.c:
+	// vm.stats.sys.v_swtch: Context switches
+	// vm.stats.sys.v_trap: Traps
+	// vm.stats.sys.v_syscall: System calls
+	// vm.stats.sys.v_intr: Device interrupts
+	// vm.stats.sys.v_soft: Software interrupts
+	// vm.stats.vm.v_forks: Number of fork() calls
+
+	return &execCollector{
+		sysctls: []bsdSysctl{
+			{
+				name:        "context_switches_total",
+				description: "Context switches",
```

Comment:
The help string is redundant and doesn't give any additional information, e.g. when this counter is being reset. I assume it starts after boot, which should be stated here and all other help strings.